core/src/main/scala/org/apache/spark/scheduler/ExecutorLossReason.scala

```diff
@@ -71,7 +71,8 @@ case class ExecutorProcessLost(
  * This is used by the task scheduler to remove state associated with the executor, but
  * not yet fail any tasks that were running in the executor before the executor is "fully" lost.
  *
- * @param workerHost it is defined when the worker is decommissioned too
+ * @param reason the reason why the executor is decommissioned
+ * @param host it is defined when the host where the executor located is decommissioned too
  */
-private [spark] case class ExecutorDecommission(workerHost: Option[String] = None)
- extends ExecutorLossReason("Executor decommission.")
+private [spark] case class ExecutorDecommission(reason: String, host: Option[String] = None)
```

Comment:
My scala knowledge is really really poor, but I would rather we make this be a non case class if you are planning to do this. Currently, I think the field "reason" is going to be duplicated in the base class ExecutorLossReason  and the ExecutorDecommission. 

That's also the reason why you are pattern matching it above with an additional _ (for the `reason`) argument, when you really don't care about the `reason`.

Reply:
The case class is neeed because we'd apply pattern matching on it.

The "reason" is necessary because of class inheritance, no? Please see `ExecutorProcessLost` for as instance, the `ExecutorProcessLost` also has the field `_message`, which is needed to assigne to  `ExecutorProcessLost.message`

Reply:
You can right unapply methods if you need to do pattern matching with something other than a case class.